analyzer/windows/modules/auxiliary/autoruns.py

```diff
@@ -0,0 +1,145 @@
+        self.output_file_start = "autoruns_start.txt"
+        self.output_file_end = "autoruns_end.txt"
+        self.output_file_diff = "autoruns.diff"
+        self.startupinfo = subprocess.STARTUPINFO()
+        self.startupinfo.dwFlags |= subprocess.STARTF_USESHOWWINDOW
+
+        if not os.path.exists(self.output_dir):
+            os.makedirs(self.output_dir)
+            #reg.exe ADD "HKCU\Software\Sysinternals\Autoruns" /v EulaAccepted /t REG_DWORD /d 1 /f
+        try:
+            with CreateKeyEx(HKEY_CURRENT_USER, "Software\Sysinternals\Autoruns", 0, KEY_ALL_ACCESS) as key:
+                SetValueEx(key, "EulaAccepted", 0, REG_DWORD, 1)
+        except OSError as e:
+            pass
+
+    def start(self):
+        # First figure out what architecture the system in running (64 or 86)
+        bin_path = os.path.join(os.getcwd(), "bin")
+
+        if "AMD64" in platform.uname():
+            autoruns = os.path.join(bin_path, "autorunsc64.exe")
+        else:
+            autoruns = os.path.join(bin_path, "autorunsc.exe")
+
+        if not os.path.exists(autoruns):
+            raise CuckooPackageError(
+                "In order to use the Autorun functionality, it "
+                "is required to have Autorunsc setup with Cape."
+            )
+        autoruns = autoruns.replace("\\","\\\\")
+        run_args = self.options.get("run_args")
+        if not run_args:
+            run_args = f"-a * -c -o {self.output_dir}\\\\{self.output_file_start}"
+
+        run_cmd = f"{autoruns} {run_args}"
+        run_cmd = shlex.split(run_cmd)
+        log.debug(run_cmd)
+        #with open(f"{self.output_dir}\\{self.output_file_start}", "w") as f:
+        #    subprocess.Popen(run_cmd, startupinfo=self.startupinfo, stdout=f, stderr=f, text=True)
+        subprocess.Popen(run_cmd, startupinfo=self.startupinfo)
+
+    def stop(self):
+
+        # First figure out what architecture the system in running (64 or 86)
+        bin_path = os.path.join(os.getcwd(), "bin")
+
+        if "AMD64" in platform.uname():
+            autoruns = os.path.join(bin_path, "autorunsc64.exe")
+        else:
+            autoruns = os.path.join(bin_path, "autorunsc.exe")
+
+        if not os.path.exists(autoruns):
+            raise CuckooPackageError(
+                "In order to use the Autorun functionality, it "
+                "is required to have Autorunsc setup with Cape."
+            )
+        autoruns = autoruns.replace("\\","\\\\")
+        run_args = self.options.get("run_args")
+        if not run_args:
+            run_args = f"-a * -c -o {self.output_dir}\\\\{self.output_file_end}"
+
+        run_cmd = f"{autoruns} {run_args}"
+        run_cmd = shlex.split(run_cmd)
+        log.debug(run_cmd)
+        #with open(f"{self.output_dir}\\{self.output_file_end}", "w") as f:
+        #    process = subprocess.Popen(run_cmd, startupinfo=self.startupinfo, stdout=f, text=True)
+        process = subprocess.Popen(run_cmd, startupinfo=self.startupinfo)
```

Comment:
since you are running the same command at `start()` and `stop()`, you should be able to just set `run_cmd` to a class variable and re-use that, rather than duplicate logic across methods

Reply:
Also I think you should use the synchronous `run()` method from `subprocess` rather than the asyncronous `Popen` + `wait()`

https://docs.python.org/3/library/subprocess.html#subprocess.run

Reply:
It is used because we are passing the flags of startupinfo, run is like a wrapper around Popen which doesn't provide that functionality.

Reply:
ah right right, carry on with Popen then!